### PR TITLE
Unit test for Image.tsx

### PR DESCRIPTION
**Pull request by namnguyen20999**
image

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 1]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
             type: "SEND_ACTION_ACTION",
         });
     });
+    it("should return [undefined, undefined, false] when content prop is not provided", () => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]:
```suggestion
    it("URL used when content prop is not provided", () => {
```

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 2]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
...
+        const { getByRole } = render(<Image defaultContent="/url/to/default/image.png" />);
+        const img = getByRole("img") as HTMLImageElement;
+        expect(img.src).toBe("http://localhost/url/to/default/image.png");
+    });
+    it("should return [undefined, undefined, false] when content prop is an empty string", () => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]:
```suggestion
    it("URL replaced when content prop is an empty string", () => {
```

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 3]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
...
+        const { getByRole } = render(<Image defaultContent="/url/to/default/image.png" content="" />);
+        const img = getByRole("img") as HTMLImageElement;
+        expect(img.src).toBe("http://localhost/");
+    });
+    it("should return [undefined, undefined, false] when content prop is a string of length less than 4", () => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]:
```suggestion
    it("URL replaced when content prop is a set", () => {
```
Or there's something I don't know

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 4]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
...
+            <Image defaultContent="/url/to/default/image.png" content="/url/to/content/image.svg" />,
+        );
+
+        // Wait for useEffect to complete
+        await new Promise((resolve) => setTimeout(resolve, 0));
+
+        const svg = container.querySelector("svg");
+        expect(svg?.outerHTML.replace(/"/g, "'")).toBe(svgContent.replace(/"/g, "'"));
+    });
+    it("should return the content prop when it is SVG XML", () => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]:
```suggestion
    it("should return the content prop when it is inline SVG", () => {
```

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 5]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
...
+        const content = "<svg xmlns='http://www.w3.org/2000/svg'></svg>";
+        const { container } = render(<Image defaultContent="/url/to/default/image.png" content={content} />);
+        const svg = container.querySelector("svg");
+        expect(svg?.outerHTML.replace(/"/g, "'")).toBe(content);
+    });
+    it("should return the content prop when it is a string of length 4 or more that is not a SVG file or SVG XML", () => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]:
```suggestion
    it("should return the content prop in the general case", () => {
```

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 6]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
...
+        const { container } = render(
+            <Image defaultContent="/url/to/default/image.png" content="/url/to/content/image.svg" />,
+        );
+        // Wait for useEffect to complete
+        await new Promise((resolve) => setTimeout(resolve, 0));
+
+        const div = container.querySelector("div");
+        expect(div).toBeInTheDocument();
+    });
+    it("should render a div when content prop is SVG XML", () => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]:
```suggestion
    it("should render a div when content prop is inline SVG", () => {
```

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 7]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
...
+        const content = "<svg xmlns='http://www.w3.org/2000/svg'></svg>";
+        const { container } = render(<Image defaultContent="/url/to/default/image.png" content={content} />);
+        const div = container.querySelector("div");
+        expect(div).toBeInTheDocument();
+    });
+    it("should render an img when content prop is not a SVG file or SVG XML", () => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]:
```suggestion
    it("should render an img when content prop is not SVG", () => {
```

**Review by FabienLelaquais — DISMISSED** [review]
Great job.
You may, or may not, clarify the test messages - I'll be happy to re-review

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Rename test

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Rename

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 3 above)
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Idem

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx [hunk 8]
```diff
@@ -79,4 +82,72 @@ describe("Image Component", () => {
...
+    it("should return the content prop when it is a SVG file", async () => {
+        const svgContent = '<svg xmlns="http://www.w3.org/2000/svg"></svg>';
+        (axios.get as jest.Mock).mockResolvedValue({ data: svgContent });
+
+        const { container } = render(
+            <Image defaultContent="/url/to/default/image.png" content="/url/to/content/image.svg" />,
+        );
+
+        // Wait for useEffect to complete
+        await new Promise((resolve) => setTimeout(resolve, 0));
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Image.spec.tsx:109]: Interesting

File: frontend/taipy-gui/src/components/Taipy/Image.tsx [hunk 9]
```diff
@@ -58,12 +58,15 @@ const Image = (props: ImageProps) => {
         return [undefined, undefined, false];
     }, [content]);
 
-    const style = useMemo(() => ({
-        width: width,
-        height: height,
-        display: inlineSvg ? "inline-flex" : undefined,
-        verticalAlign: inlineSvg ? "middle" : undefined
-    }), [width, height, inlineSvg]);
+    const style = useMemo(
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Image.tsx:61]: Still formatting differences!

File: frontend/taipy-gui/src/components/Taipy/Image.tsx [hunk 10]
```diff
@@ -76,21 +79,23 @@ const Image = (props: ImageProps) => {
...
-                    disabled={!active}
-                    title={label}
-                >
-                    {inlineSvg ? (
-                        <div ref={divRef} style={style} />
-                    ) : (
-                        <img src={content} style={style} alt={label} />
-                    )}
-                </Button>
+                <span>
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Image.tsx:82]: Good catch

**Review by FredLL-Avaiga — DISMISSED** [review]

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 1 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 3 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 2 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 4 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 5 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 7 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 6 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 2 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 1 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.spec.tsx (hunk 3 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.spec.tsx]: Updated

File: frontend/taipy-gui/src/components/Taipy/Image.tsx (hunk 9 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Image.tsx:61]: Might need to add a prettierrc file so that we could have a common formatting style

**Review by FabienLelaquais — APPROVED** [review]
👌👌

**Pull request merged**